test_requests.py

```diff
@@ -1224,6 +1224,7 @@ def test_equality(self):
         del othercid['spam']
         assert cid != othercid
         assert cid == {'spam': 'blueval', 'eggs': 'redval'}
+        assert cid.__eq__(object()) == NotImplemented
```

Comment:
Shouldn't this be

```py
with pytest.raises(NotImplementedError):
    cid == object()
```

We shouldn't be directly invoking dunder methods anywhere in the code base, not even tests.